odbc/src/api/data_tests.rs

```diff
@@ -551,6 +551,128 @@ mod unit {
         }
     }
 
+    #[test]
+    fn sql_get_binary_data() {
+        use crate::api::functions::SQLGetData;
+        use odbc_sys::CDataType;
+        let mut stmt = Statement::with_state(std::ptr::null_mut(), StatementState::Allocated);
+        stmt.mongo_statement = Some(Box::new((*MQ).clone()));
+        let stmt_handle: *mut _ = &mut MongoHandle::Statement(RwLock::new(stmt));
+        unsafe {
+            assert_eq!(SqlReturn::SUCCESS, SQLFetch(stmt_handle as *mut _,));
+            let buffer: *mut std::ffi::c_void = Box::into_raw(Box::new([0u8; 200])) as *mut _;
+            let buffer_length: isize = 100;
```

Comment:
Do you think we should make the buffer length specifiable to verify expected failure behavior?

From the docs for date to bin:
SQL_C_BINARY	Byte length of data <= BufferLength

|C type identifier|Test|**TargetValuePtr*|**StrLen_or_IndPtr*|SQLSTATE|  
|-----------------------|----------|------------------------|----------------------------|--------------| 
|SQL_C_BINARY|Byte length of data <= *BufferLength*<br /><br /> Byte length of data > *BufferLength*|Data<br /><br /> Undefined|Length of data in bytes<br /><br /> Undefined|n/a<br /><br /> 22003|

Reply:
I see now that you shouldn't actually be able to get a `date_time` as binary in pieces. But it says the output is undefined, so I think our ability to get them by pieces is fine. It only complicates the code needlessly to provide _less_ useful behavior to our clients. This is kind of an odd part of the spec. Thanks for pointing it out.

What do you think @nbagnard?

Reply:
The output is undefined but the SQLstate is 22003 and SQLERROR.
I think at this point because this was an error, SQLGetData will return Invalid cursor state when called again.
Check with a working driver.
Generally speaking, streaming is for SQL Binary and Character data and SQLGetData returns 01004 and SUCCESS WITH INFO when you can call SQLGetData again.

Reply:
But I think Nathan as a point and we don't seem to have a test checking that we get an error when the buffer is too small. Maybe I missed it, totally possible.

Reply:
Because we don't return an error. I let it run like all other binary data.